### ISSUE TITLE
Inserting a row after sorting and loadData breaks index translation and results in additional blank rows

### ISSUE THREAD

**Issue by ngxchinhs** [handsontable/handsontable#3809]
Hello all,

I have a problem when alter "insert_row". Expected is "insert only one row at the bottom of the table"
But there are two new row are inserted at two position, one is at the top and the second is the bottom of table.

Live example on JSFiddle
http://jsfiddle.net/01xy0kap/1/
1. Click button AddRow to add any row.
2. Click sort column (descending)
3. Click LoadData button
4. Click AddRow, the problem wil be occurred..

**Comment by AMBudnik** [comment]
Thanks for sharing @ngxchinhs 
I got the same issue, at the end we have 2 new rows.

**Comment by budnix** [comment]
Connected with https://github.com/handsontable/handsontable/issues/3849.

**Comment by AMBudnik** [comment]
Updated demo: http://jsfiddle.net/handsoncode/mewuug01/ (vanilla version).

We get the new row at the top of the table when it should be added at the bottom.

**Comment by AMBudnik** [comment]
Still replicable in 0.31.2

**Comment by AMBudnik** [comment]
Works the same in v0.35.0

**Comment by AMBudnik** [comment]
The issue is still replicable in v 0.38.1/1.18.0

**Comment by aninde** [comment]
This bug is still replicable at current version 7.1.1.

New `index mapper` fixed inserting data after `sorting` and `loadData`. It will be released soon.

We are currently testing a version with many breaking changes, I recommend observing information from issue [#5751.](https://github.com/handsontable/handsontable/issues/5751)

**Demo**
version 8.0.0.rc http://jsfiddle.net/aninde/L42pjwfh/ 
Version 8.0.0rc http://jsfiddle.net/aninde/908vdprx/

**Environment**
Handsontable version:8.0.0.rc

Browser Name and version: Google Chrome version 76.0.3809.100
Operating System: macOS Mojave 10.14.6

**Comment by AMBudnik** [comment]
Hey @ngxchinhs 

now you can try our official 8.0.0-beta1 where the issue is solved http://jsfiddle.net/m8b1npvt/

```html
</style><script src="https://cdn.jsdelivr.net/gh/handsontable/handsontable@release/8.0.0-beta.1/dist/handsontable.full.js"></script><link type="text/css" rel="stylesheet" href="https://cdn.jsdelivr.net/gh/handsontable/handsontable@release/8.0.0-beta.1/dist/handsontable.full.css">
```
`npm i handsontable@beta`

**Comment by pnowak** [comment]
Hi @ngxchinhs 

The issue in 8.0.0-rev1 version is still solved - https://jsfiddle.net/wy9gcoqu/

**Comment by AMBudnik** [comment]
Thank you for checking this @pnowak 
@@ngxchinhs We can close the issue as solved in v8.0.0

CDN https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.css
https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.js
NPM npm i handsontable
Release Notes https://github.com/handsontable/handsontable/releases
